### PR TITLE
Reese refactor

### PR DESCRIPTION
**Pull request by Bensonre**
everything still works, tests all pass. I added variable names I hope this more clear and easier to read.

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -27,62 +27,80 @@ public boolean alreadyPlaced(Ship ship) {
...
 
+	/*
+	DO NOT change the signature of this method. It is used by the grading scripts.
+	 */
+	public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
+		if (alreadyPlaced(ship))
+			return false;
+		int shipLength = ship.getLength(); // store length in var for easier access
+		if (isVertical) { // test cases if the ship is vertical
+			return (whenvert(ship, shipLength, x, y, isVertical));
```
Comment (xMestas) [inline src/main/java/cs361/battleships/models/Board.java]: Good, splitting the two cases into their own functions makes it easier to follow.  I recommend naming the functions placeVertical and placeHorizontal.

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -27,62 +27,80 @@ public boolean alreadyPlaced(Ship ship) {
...
+		} else {// the following works the same as the above code but if the ship is horizontal
```
Comment (xMestas) [inline src/main/java/cs361/battleships/models/Board.java]: Update this comment

File: src/main/java/cs361/battleships/models/Board.java [hunk 3]
```diff
@@ -27,62 +27,80 @@ public boolean alreadyPlaced(Ship ship) {
...
-							if (x + k == this.ships.get(i).getOccupiedSquares().get(j).getRow()) {// tests to see if the ship is in the same row as the ith ship
-								if (y == this.ships.get(i).getOccupiedSquares().get(j).getColumn()) {// tests to see if the ship is in the same col as the ith ship
-									return false;// if both the above cases are true the ship is in the same square as the ith ship and can't be placed.
-								}
+// called when the ship is verticle and being placed
+// returns weather of not ship can be placed
+	public boolean whenvert(Ship ship, int shipLength, int x, char y, boolean isVertical) {
+		if (x + (shipLength) > 11 || x < 1)
+			return false; // ensures ship cant go over the edge of board
+		else {
```
Comment (xMestas) [inline src/main/java/cs361/battleships/models/Board.java]: No need for an else as the function will die if it goes into that if statement anyways.

File: src/main/java/cs361/battleships/models/Board.java [hunk 4]
```diff
@@ -27,62 +27,80 @@ public boolean alreadyPlaced(Ship ship) {
...
+			if (y >= 'K' || y < 'A') {
+				return false; // ships can't go over edge of board 74 == J
+			}
+			List<Ship> shipsList = this.ships;
+			for (int i = 0; i < shipsList.size(); i++) {// this loop ensure a ships isn't placed over a different one
+				//i represent the ship being acessed in ships list, the loop itterates through the number of ships
+				Ship currentShip = shipsList.get(i);
+				for (int j = 0; j < currentShip.getLength(); j++) {//j represents the the square being checked in ship i's square list
```
Comment (xMestas) [inline src/main/java/cs361/battleships/models/Board.java]: Still a lot, but much easier to follow now

File: src/main/java/cs361/battleships/models/Board.java [hunk 5]
```diff
@@ -27,62 +27,80 @@ public boolean alreadyPlaced(Ship ship) {
...
 
-				for (int i = 0; i < this.ships.size(); i++) {
-					for (int j = 0; j < this.ships.get(i).getLength(); j++) {
-						for (int k = 0; k < shipLength; k++) {
-							if (y + k == this.ships.get(i).getOccupiedSquares().get(j).getColumn()) {
-								if (x == this.ships.get(i).getOccupiedSquares().get(j).getRow()) {
-									return false;
-								}
+			List<Ship> shipsList = this.ships; // These loops work the same as whenvert but with moves across col instead of rows
+			for (int i = 0; i < shipsList.size(); i++) {
```
Comment (xMestas) [inline src/main/java/cs361/battleships/models/Board.java]: Comments here too>

**Comment by Bensonre** [comment]
fixed stuff philly asked for

**Pull request merged**